.github/workflows/publish.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  push:
+    branches:
+      - main
+      - ci
+  schedule:
+    - cron: 0 0 * * *
```

Comment:
I suggest building each day slightly after midnight so that we don't have race conditions. We might also want to work out what happens during BST -- whether this ends up being the previous day.